forum/qa-plugin/orginal-images/original-images-page.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Xandros15;
+
+
+class original_images_page
+{
+    const REQUEST_NAME = 'wysiwyg-editor-upload';
+    const MAX_FILE_SIZE = 1024 * 1024 * 3; //3MB
+    const UPLOAD_FUNCTIONS_FILE = \QA_INCLUDE_DIR . 'qa-app-upload.php';
+
+    const CKE_OPT_UPLOAD_IMAGES = 'ckeditor4_upload_images';
+    const CKE_OPT_UPLOAD_ALL = 'ckeditor4_upload_all';
+
+    const QA_OPT_ONLY_IMAGE = 'qa_only_image';
+
+    /**
+     * @inheritdoc
+     */
+    public function match_request($request)
+    {
+        return $request == self::REQUEST_NAME;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    function process_request($request)
+    {
+        //@todo remove it after they fix multiple request for one pages
```

Comment:
what to do, what to do...

Cała aplikacja nie zachowuje się logicznie. Pod jeden endpoint moge podpiąć JEDEN process_request i zależy tylko od tego, jaką nazwe ma dany plugin (im jest wyżej w tablicy, to jest ładowany, a potem pętla nie idzie dalej).

Jeśli to zmienią, to też trzeba będzie zmienić.